packages/ra-ui-materialui/src/field/ArrayField.spec.tsx

```diff
@@ -158,4 +174,41 @@ describe('<ArrayField />', () => {
             ).toBeTruthy();
         });
     });
+
+    it('should not select the same id in both ArrayFields when selected in one', async () => {
+        render(<TwoArrayFieldsSelection />);
+        await waitFor(() => {
+            expect(screen.queryAllByRole('checkbox').length).toBeGreaterThan(2);
+        });
+
+        const checkboxes = screen.queryAllByRole('checkbox');
+
+        expect(checkboxes.length).toBeGreaterThan(3);
```

Comment:
By reading the test, I don't know why you wait for that. If you want to wait for the two lists to render, please use a `findByText` looking for the actual array values instead.